src/azul/service/repository_file_service.py

```diff
@@ -0,0 +1,122 @@
+from functools import (
+    lru_cache,
+)
+from time import (
+    time,
+)
+from typing import (
+    Optional,
+)
+
+from chalice import (
+    NotFoundError,
+)
+from elasticsearch_dsl.response import (
+    Hit,
+)
+from furl import (
+    furl,
+)
+# noinspection PyPackageRequirements
+import google.cloud.storage as gcs
+from more_itertools import (
+    one,
+)
+
+from azul import (
+    CatalogName,
+)
+from azul.drs import (
+    DRSClient,
+)
+from azul.dss import (
+    shared_credentials,
+)
+from azul.plugins import (
+    RepositoryPlugin,
+)
+from azul.service.elasticsearch_service import (
+    ElasticsearchService,
+)
+from azul.tdr import (
+    TDRClient,
+)
+from azul.types import (
+    JSON,
+)
+
+
+class RepositoryFileService(ElasticsearchService):
+
+    # FIXME: rename class to RepositoryService
+    #        https://github.com/DataBiosphere/azul/issues/2277
+
+    def __init__(self) -> None:
+        super().__init__()
+
+    @lru_cache(maxsize=None)
+    def repository_plugin(self, catalog: CatalogName) -> RepositoryPlugin:
+        return RepositoryPlugin.load(catalog).create(catalog)
+
+    def _hit_to_doc(self, catalog: CatalogName, hit: Hit) -> JSON:
+        return self.translate_fields(catalog, hit.to_dict(), forward=False)
+
+    def _get_file_document(self,
+                           catalog: CatalogName,
+                           file_uuid: str,
+                           file_version: Optional[str]) -> JSON:
+        """
+        Return a file document from Elasticsearch.
+        """
+        filters = {
+            "fileId": {"is": [file_uuid]},
+            **({"fileVersion": {"is": [file_version]}}
+               if file_version else {})
+        }
+        es_search = self._create_request(catalog=catalog,
+                                         filters=filters,
+                                         post_filter=True,
+                                         enable_aggregation=False,
+                                         entity_type='files')
+        hit = one(self._hit_to_doc(catalog, hit) for hit in es_search.scan())
+        if not hit:
+            raise NotFoundError(f'Unable to find file {catalog}/{file_uuid}/{file_version}')
```

Comment:
I don't think this slash notation has any precedent. We can't just invent notations. Also think how this looks if file_version is None.